corehq/apps/app_manager/static/app_manager/js/form_workflow.js

```diff
@@ -0,0 +1,107 @@
+/* global ko, _, $ */
+
+(function() {
+    'use strict';
+
+    window.FormWorkflow = function(options) {
+        var self = this;
+
+        // Human readable labels for the workflow types
+        self.labels = options.labels;
+
+        // Workflow type. See FormWorkflow.Values for available types
+        self.workflow = ko.observable(options.workflow);
+
+        self.locale = options.locale || 'en';
+
+        // Element used to trigger a change when form link is removed
+        self.$changeEl = options.$changeEl || $('#form-workflow .workflow-change-trigger');
+
+        /* Form linking */
+        self.showFormLinkUI = ko.observable(self.workflow() === FormWorkflow.Values.FORM);
+        self.forms = _.map(options.forms, function(f) {
+            return new FormWorkflow.Form(f, { locale: self.locale });
+        });
+        self.formLinks = ko.observableArray(_.map(options.formLinks, function(link) {
+            return new FormWorkflow.FormLink(link.xpath, link.form_id, { forms: self.forms });
+        }));
+    };
+
+    FormWorkflow.Values = {
+        DEFAULT: 'default',
+        ROOT: 'root',
+        MODULE: 'module',
+        PREVIOUS_SCREEN: 'previous_screen',
+        FORM: 'form',
+    };
+
+    FormWorkflow.Errors = {
+        FORM_NOTFOUND: 'This form either no longer exists or has a different case type',
+    };
+
+
+    FormWorkflow.prototype.workflowDisplay = function() {
```

Comment:
This could be a `computed observable`: http://knockoutjs.com/documentation/computedObservables.html